include/stdx/udls.hpp

```diff
@@ -11,12 +11,26 @@ namespace stdx {
 inline namespace v1 {
 
 namespace detail {
+template <char C>
+constexpr static bool is_decimal_digit_v = C >= '0' and C <= '9';
+template <char C> constexpr static bool is_digit_sep_v = C == '\'';
+
+template <char C, typename Sum> CONSTEVAL auto maybe_add_digit(Sum s) {
+    if constexpr (is_decimal_digit_v<C>) {
+        s *= 10;
+        s += C - '0';
+    }
+    return s;
+}
+
 template <typename T, char... Chars> CONSTEVAL auto decimal() -> T {
-    static_assert((... and (Chars >= '0' and Chars <= '9')),
-                  "decimal numbers only are supported");
+    static_assert(
+        (... and (is_decimal_digit_v<Chars> or is_digit_sep_v<Chars>)),
+        "decimal numbers only are supported");
```

Comment:
issue: octal numbers are indicated with a `0` followed by one or more octal digits. this would interpret an octal number as decimal.

quibble: the message should read something like `"whole decimal numbers only are supported"`

awesomeness: use `STATIC_ASSERT` and show the invalid character string passed to the udl! 😎

Reply:
OK, do we want to support hex literals also? I'll beef this up.

Reply:
In fact, our own `static_assert`s are not necessary... invalid literals are detected by the compiler before it even parses the UDL suffix and hits our code.

Reply:
If we implement all the integer base representations supported by C++, then yes, that would be the best option.